### PR TITLE
Add gitleaks doc

### PR DESCRIPTION
**Pull request by stone-z**
Towards https://github.com/giantswarm/giantswarm/issues/7657
This behavior will be added by https://github.com/giantswarm/devctl/pull/100, pending some trials by Ludacris. We won't merge this until the `devctl` PR is merged

### CODE REVIEW

File: go/general_style_guide.md [hunk 1]
```diff
@@ -117,7 +123,75 @@ CVE-2020-15114 until=2020-10-01
...
+    '''(go.mod|go.sum)$''']
+
+    # Exclude anything in vendor/
+    paths = [ '''vendor\/.*''']
+
+    # Exclude a certain value
+    regexes = ['''MyPublicOkToCheckInTokenValue''']
+
+    # Exclude a file by adding it to the existing default 'files' list, e.g.:
+    files = ['''^this-file-is-all-high-entropy-strings.json$'''
```
Comment (tfussell) [inline go/general_style_guide.md:173]:
```suggestion
    files = ['''^this-file-is-all-high-entropy-strings.json$''',
```

File: go/general_style_guide.md [hunk 2]
```diff
@@ -40,6 +40,12 @@ This should be used sparingly to avoid forming a bad habit of simply silencing b
 Details for using all of these methods can be found in the
 [False Positives](https://github.com/golangci/golangci-lint#false-positives) section of the docs.
 
+### Tooling
+
+You can lint your code straight from the editor
+
+- [Intellij](./intellij.md)
```
Comment (tfussell) [inline go/general_style_guide.md:47]:
```suggestion
- [IntelliJ](./intellij.md)
```

**Review by MarcelMue — APPROVED** [review]

**Review by tfussell — APPROVED** [review]

**Comment by stone-z** [comment]
@giantswarm/team-fusion-engineers anybody up for a 3rd review?

File: go/general_style_guide.md [hunk 3]
```diff
@@ -117,7 +123,75 @@ CVE-2020-15114 until=2020-10-01
...
+
+### Disabling for a Repository
+
+If it is simply too noisy to use the scanner on a particular repository, it is possible to exclude the workflow.
+
+If you are generating workflows with `devctl`, you can use the `--check-secrets` flag to control the workflow:
+
+```shell
+devctl gen workflows --flavour <flavour> --check-secrets=false
+```
```
Comment (kopiczko) [inline go/general_style_guide.md:187]: That should not be the case IMO. I would liketo stop recommending using `devctl gen workflows` directly.

**Review by kopiczko — APPROVED** [review]

File: go/general_style_guide.md (hunk 3 above)
Comment (stone-z) [reply go/general_style_guide.md:187]: Oh ok, that's news to me as well. Would you rather I remove it?

Comment (kopiczko) [reply go/general_style_guide.md:187]: Yes.

Comment (stone-z) [reply go/general_style_guide.md]: Gone

**Pull request merged**